bottlecap/src/proc/mod.rs

```diff
@@ -180,27 +212,154 @@ fn get_uptime_from_path(path: &str) -> Result<f64, io::Error> {
+                    fd_max = fd_max.min(fd_max_pid);
+                } else {
+                    debug!("File descriptor max data not found in file {}", limits_path);
+                }
+                break;
+            }
+        }
+    }
+
+    fd_max
+}
+
+pub fn get_fd_use_data(pids: &[i64]) -> Result<f64, io::Error> {
+    get_fd_use_data_from_path(PROC_PATH, pids)
+}
+
+fn get_fd_use_data_from_path(path: &str, pids: &[i64]) -> Result<f64, io::Error> {
+    let mut fd_use = 0;
+
+    for &pid in pids {
+        let fd_path = format!("{path}/{pid}/fd");
+        let Ok(files) = fs::read_dir(fd_path) else {
+            return Err(io::Error::new(
+                io::ErrorKind::InvalidData,
+                "File descriptor use data not found",
+            ));
+        };
+        let count = files.count();
+        fd_use += count;
+    }
+
+    Ok(fd_use as f64)
+}
+
+#[must_use]
+pub fn get_threads_max_data(pids: &[i64]) -> f64 {
+    get_threads_max_data_from_path(PROC_PATH, pids)
+}
+
+fn get_threads_max_data_from_path(path: &str, pids: &[i64]) -> f64 {
+    let mut threads_max = constants::LAMBDA_EXECUTION_PROCESSES_DEFAULT_LIMIT;
+    // regex to capture the soft limit value (first numeric value after the title)
+    let re = Regex::new(r"^Max processes\s+(\d+)").expect("Failed to create regex");
+
+    for &pid in pids {
+        let limits_path = format!("{path}/{pid}/limits");
+        let Ok(file) = File::open(&limits_path) else {
+            continue;
+        };
+
+        let reader = io::BufReader::new(file);
+        for line in reader.lines().map_while(Result::ok) {
+            if let Some(line_items) = re.captures(&line) {
+                if let Ok(threads_max_pid) = line_items[1].parse() {
+                    threads_max = threads_max.min(threads_max_pid);
+                } else {
+                    debug!("Threads max data not found in file {}", limits_path);
+                }
+                break;
+            }
+        }
+    }
+
+    threads_max
+}
+
+pub fn get_threads_use_data(pids: &[i64]) -> Result<f64, io::Error> {
+    get_threads_use_data_from_path(PROC_PATH, pids)
+}
+
+fn get_threads_use_data_from_path(path: &str, pids: &[i64]) -> Result<f64, io::Error> {
+    let mut threads_use = 0;
+
+    for &pid in pids {
+        let task_path = format!("{path}/{pid}/task");
+        let Ok(files) = fs::read_dir(task_path) else {
+            return Err(io::Error::new(
+                io::ErrorKind::InvalidData,
+                "Threads use data not found",
+            ));
+        };
+
+        threads_use += files
+            .flatten()
+            .filter_map(|dir_entry| dir_entry.file_type().ok())
+            .filter(fs::FileType::is_dir)
+            .count();
+    }
+
+    Ok(threads_use as f64)
+}
+
 #[cfg(test)]
 #[allow(clippy::unwrap_used)]
 mod tests {
     use super::*;
     use std::path::PathBuf;
 
+    const DELTA: f64 = 1e-6;
```

Comment:
Isn't this just a precision measure? Maybe DELTA is confusing to the reader

Reply:
Hmm should I switch it back to how it was earlier?

Reply:
No, I mean, rename the variable to something like `PRECISION` its much clearer – sorry, just a nit